core/src/main/python/synapse/ml/automl/HyperparamBuilder.py

```diff
@@ -39,50 +41,72 @@ def build(self):
         """
         return self.hyperparams.items()
 
+
 class DiscreteHyperParam(object):
     """
     Specifies a discrete list of values.
     """
+
     def __init__(self, values, seed=0):
         ctx = SparkContext.getOrCreate()
         self.jvm = ctx.getOrCreate()._jvm
-        self.hyperParam = self.jvm.com.microsoft.azure.synapse.ml.automl.HyperParamUtils.getDiscreteHyperParam(values, seed)
+        self.hyperParam = self.jvm.com.microsoft.azure.synapse.ml.automl.HyperParamUtils.getDiscreteHyperParam(
+            values, seed
+        )
 
     def get(self):
         return self.hyperParam
 
+
 class RangeHyperParam(object):
     """
     Specifies a range of values.
     """
+
     def __init__(self, min, max, seed=0):
         ctx = SparkContext.getOrCreate()
         self.jvm = ctx.getOrCreate()._jvm
-        self.rangeParam = self.jvm.com.microsoft.azure.synapse.ml.automl.HyperParamUtils.getRangeHyperParam(min, max, seed)
+        self.rangeParam = self.jvm.com.microsoft.azure.synapse.ml.automl.HyperParamUtils.getRangeHyperParam(
+            min, max, seed
+        )
 
     def get(self):
         return self.rangeParam
 
+
 class GridSpace(object):
     """
     Specifies a predetermined grid of values to search through.
     """
+
     def __init__(self, paramValues):
         ctx = SparkContext.getOrCreate()
         self.jvm = ctx.getOrCreate()._jvm
-        hyperparamBuilder = self.jvm.com.microsoft.azure.synapse.ml.automl.HyperparamBuilder()
+        hyperparamBuilder = self.jvm.org.apache.spark.ml.tuning.ParamGridBuilder()
```

Comment:
I'm a bit confused, I think we should still be using our HyperparamBuilder here and not spark's ParamGridBuilder?

Reply:
The reason is that GridSpace constructor accepts `Array[ParamMap]`, which is exactly what `ParamGridBuilder.build()` function returns. `HyperparamBuilder.build()` returns `Array[Param[_], Dist[_]]` instead. Technically I can still use HyperparamBuilder and convert the result, but it's only used as a local variable anyway, so I don't think it matters.

Reply:
"Technically I can still use HyperparamBuilder and convert the result"

hmm I can't recall why I wrote this way in the original PR, it was 4 years ago:

https://github.com/microsoft/SynapseML/pull/168

Also I don't see any scala test for gridparam which is strange.  I do think that the intention was to always use HyperparamBuilder.  Let me check.

I also don't understand this:
```
if not isinstance(hyperparam, DiscreteHyperParam)
```
random param should be supported as well, and we get the value when adding it to the grid.  So I'm confused why it doesn't work.

Reply:
hmm, ok, maybe this can be merged as is and I can try and look into whether it's possible to use HyperparamBuilder still, support RandomParam in addition to DiscreteHyperParam and also add some tests for this workflow, as I don't actually see this GridSpace tested anywhere.